llvm/lib/Target/ARM/AsmParser/ARMAsmParser.cpp

```diff
@@ -6886,19 +6904,17 @@ void ARMAsmParser::fixupGNULDRDAlias(StringRef Mnemonic,
 // operand. If the conversion fails an error is diagnosed, and the function
 // returns true.
 bool ARMAsmParser::CDEConvertDualRegOperand(StringRef Mnemonic,
-                                            OperandVector &Operands) {
+                                            OperandVector &Operands,
+                                            unsigned MnemonicOpsEndInd) {
   assert(MS.isCDEDualRegInstr(Mnemonic));
-  bool isPredicable =
-      Mnemonic == "cx1da" || Mnemonic == "cx2da" || Mnemonic == "cx3da";
-  size_t NumPredOps = isPredicable ? 1 : 0;
 
-  if (Operands.size() <= 3 + NumPredOps)
+  if (Operands.size() < 2 + MnemonicOpsEndInd)
```

Comment:
Should this be a `<=` instead of `<`? Further down, line 6952, you're dereferencing `Operands[MnemonicOpsEndInd + 2]`, which would be one beyond the end if `Operands.size()` was _exactly_ the value you're comparing it with here.

(The previous code is `<=`, which was what caught my eye)

Reply:
Ahh yup, thank you